CRM/Member/BAO/Membership.php

```diff
@@ -1594,17 +1573,35 @@ public static function buildMembershipTypeValues(&$form, $membershipTypeID = arr
       'max_related',
       'duration_unit',
       'duration_interval',
-    );
+    ];
 
-    while ($dao->fetch()) {
-      $membershipTypeValues[$dao->id] = array();
-      foreach ($membershipTypeFields as $mtField) {
-        $membershipTypeValues[$dao->id][$mtField] = $dao->$mtField;
-      }
+    $membershipTypeParams = [
+      'is_active' => $activeOnly,
```

Comment:
this is the cause of the test fails - this works

```
--- a/CRM/Member/BAO/Membership.php
+++ b/CRM/Member/BAO/Membership.php
@@ -1585,10 +1585,12 @@ WHERE  civicrm_membership.contact_id = civicrm_contact.id
     ];
 
     $membershipTypeParams = [
-      'is_active' => $activeOnly,
       'return' => $membershipTypeFields,
       'options' => ['limit' => 0],
     ];
+    if ($activeOnly) {
+      $membershipTypeParams['is_active'] = 1;
+    }
     if ($financialTypes) {
       $membershipTypeParams['financial_type_id'] = ['IN' => array_keys($financialTypes)];
     }
~
~

```

Reply:
fixed